akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/ClusterStatusCheck.scala

```diff
@@ -23,15 +23,22 @@ import scala.concurrent.{ ExecutionContext, Future }
 
 class ClusterStatusCheck extends ReadinessCheck with HealthCheck {
   def healthy(actorSystem: ExtendedActorSystem)(implicit ec: ExecutionContext): Future[Boolean] =
-    isUp(actorSystem)
+    status(actorSystem) map {
+      case MemberStatus.Joining  => true
+      case MemberStatus.WeaklyUp => true
+      case MemberStatus.Up       => true
+      case MemberStatus.Leaving  => false
```

Comment:
does this come a k8s liveness probe? Do we want to kill a container that is gracefully leaving? Maybe we should just ensure CoordinatedShutdown is correctly configured

Reply:
See https://github.com/akka/akka-management/issues/259 for the debate 🙂

Reply:
I guess removed is safe to blast away

Reply:
What do you mean?

Reply:
leader puts it there so it has gracefully left the cluster

Full leaving and down this process still has to happen.

I think the safest is to only get k8s involved if it is Removed. Allow the shutdown process to finish inside Akka for Leaving/Exciting.

Reply:
I see. I'll let k8s leave Akka to its excitement.